app/controllers/pages_controller.rb

```diff
@@ -7,6 +7,7 @@ def privacy_policy
     policy_id = params[:id]
 
     @privacy_policy = if policy_id
+
```

Comment:
Why do we need to set the session here? Is it not set before we redirect the user anyway? also, random white space...

Reply:
I might be wrong but the first time we offer the privacy policy id from _identity.html.erb, so it the session variable is set from the API call in that view. The second time is the accept_privacy_policy page, so the session variable is set if they have viewed the privacy page from the _identity.html.erb link, if not it is set when they view the privacy page from the accept_privacy_policy link

Reply:
I don't think the `PagesController` should be concerned/know about the session; it should be set as part of the step that links out to the policy from the wizard -- it's bleeding out into other areas of the app here IMO

Reply:
We can only access the session directly in a view or a controller, not quite sure what you mean.

Reply:
Sorry, I’m doing a bad job of explaining this — I’ll grab you tomorrow morning and we can jump on a call 👍

Reply:
On holiday til Wednesday.

Reply:
Ah ok, in that case I'll try and write it down so I don't forget:

I believe the way it works on the privacy policy step is that we render a hidden field containing the `accepted_policy_id` and that gets posted back to the API and subsequently set in the session (which is how all the other steps work in general). The thing I don't understand here is that we've now also got similar logic in the `PagesController`... why not be consistent and add a hidden field to the `identity` step as well, so you can remove this from the `PagesController`?

https://github.com/DFE-Digital/get-teacher-training-adviser-service/blob/master/app/models/accept_privacy_policy.rb#L2

https://github.com/DFE-Digital/get-teacher-training-adviser-service/blob/master/app/models/identity.rb#L2-L4

Reply:
The pages work differently, the accept_privacy_policy model has :accepted_policy_id as an attribute, and a link, the identity page just has a link. I have been back and forth over it, deleting the session variable between views etc, and as far as I can see it works perfectly. 
Both pages use `<%= link_to "privacy notice", privacy_policy_path(:id => session[:privacy_policy_id].presence || ApiClient.get_latest_privacy_policy.id) %>` . There is the session[:privacy_policy_id] which we use in the views and session[:registration]["accepted_policy_id"] which we send to the api. I see what you mean, however I don't think it should be the responsibility of the Identity model to set the accepted_policy_id attribute.